Implement profiling subcommands

nnethercote opened the pull request:
perf-record, Cachegrind, and DHAT are the profilers I've used in the past on rustc. This branch adds support for running each of them on the rustc-benchmarks.

Mark-Simulacrum commented on collector/src/bin/rustc-fake.rs:
```diff
@@ -13,24 +13,74 @@ fn main() {
+                assert!(has_valgrind);
+                cmd.arg("--tool=cachegrind")
+                    .arg("--cache-sim=no")
+                    .arg("--branch-sim=no")
```
I see "Note that you cannot specify --cache-sim=no and --branch-sim=no together, as that would leave Cachegrind with no information to collect." in the manpage for valgrind; is that information wrong/misleading?

nnethercote commented:
Well spotted on the Cachegrind docs! That info is indeed wrong, with --cache-sim=no and --branch-sim=no it still collects instruction counts. That's by far the most useful stat, and disabling the cache-sim and branch-sim make it run faster. I've added a comment about this, and I told the Valgrind maintainer about the doc error.

I also changed the `out_file` closure to use `format!` instead of multiple `push_str()` calls.
